starknet-foundry/crates/cheatable-starknet/src/rpc.rs

```diff
@@ -96,7 +99,16 @@ pub fn call_contract(
     )?)?);
     let entry_point_selector =
         EntryPointSelector(StarkHash::new(entry_point_selector.to_be_bytes())?);
-    let account_address = ContractAddress(patricia_key!(TEST_ACCOUNT_CONTRACT_ADDRESS));
+
+    let account_address = if cheated_state
+        .pranked_contracts
+        .contains_key(&contract_address)
+    {
+        cheated_state.pranked_contracts[&contract_address]
+    } else {
+        ContractAddress(patricia_key!(TEST_ACCOUNT_CONTRACT_ADDRESS))
+    };
```

Comment:
It generally should work but I wonder if it wouldn't be better if we had consistent implementation for all of cheatcodes inside syscall handler 🤔 On the other hand it would be more complicated to implement WDYT @Arcticae ?

Reply:
Oh nvm it will not work as we cannot use call_contract for proxy as it uses `TEST_ACCOUNT_CONTRACT_ADDRESS` as caller always